pdf/core/security/crypt/filter_aesv3.go

```diff
@@ -0,0 +1,173 @@
+package crypt
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"fmt"
+	"io"
+
+	"github.com/unidoc/unidoc/common"
+)
+
+func init() {
+	registerFilter("AESV3", newFilterAESV3)
+}
+
+// NewFilterAESV3 creates an AES-based filter with a 256 bit key (AESV3).
+func NewFilterAESV3() Filter {
+	f, err := newFilterAESV3(FilterDict{})
+	if err != nil {
+		panic(err)
+	}
+	return f
+}
+
+func newFilterAESV3(d FilterDict) (Filter, error) {
+	if d.Length != 0 && d.Length != 32 {
+		return nil, fmt.Errorf("Invalid AESV3 crypt filter length (%d)", d.Length)
+	}
+	return filterAESV3{}, nil
+}
+
+// filterAES implements a generic AES encryption and decryption algorithm used by AESV2 and AESV3 filter methods.
+type filterAES struct{}
+
+func (filterAES) EncryptBytes(buf []byte, okey []byte) ([]byte, error) {
```

Comment:
godoc on exported symbol

Reply:
It can be considered an unexported symbol since it's defined on unexported type. But I added a godoc anyway.